app/services/editions/db/CollectionsQueries.scala

```diff
@@ -143,6 +144,32 @@ trait CollectionsQueries {
     updatedCollections.head
   }
 
+  /**
+    * Update the indices for a list of collections, setting their index as their position in the given list.
+    *
+    * @param collection
+    * @return
+    */
+  protected def updateCollectionIndices(collectionIds: List[String])(implicit session: DBSession): Either[Error, Unit] =
+  Try {
+    collectionIds match {
+      case Nil => Right(())
+      case ids =>
+        sql"""
+          UPDATE collections
+          SET index=CASE
+            ${sqls.join(collectionIds.zipWithIndex.map {
+              case (id, index) => sqls"""WHEN id=${id} THEN ${index}"""
+            }, sqls.empty)}
```

Comment:
I am not familiar with how collections are currently ordered - it seems possible that this could re-order _any_ of them, depending on what the order of the incoming `collectionIds` is?

Reply:
When you say any – do you mean any collectionId in the list (which is the intent), or any collectionId ever? The `WHERE` on L165 should prevent the latter.

Reply:
👍